zipline/utils/events.py

```diff
@@ -539,7 +539,7 @@ def get_first_trading_day_of_month(self, dt):
         self.month = dt.month
 
         dt = dt.replace(day=1)
-        first_day = (dt if self.cal.is_open_on_day(dt)
+        first_day = (normalize_date(dt) if self.cal.is_open_on_day(dt)
```

Comment:
`normalize_date` will be wrong here if we're using a calendar that's not NYSE and US/Eastern. cc @jbredeche this is one of the cases we talked about yesterday.

In general, we need a method on `ExchangeCalendar` that says "give me the session date associated with the current minute", which doesn't appear to be available currently.